channels/pjsip/dialplan_functions.c

```diff
@@ -1784,3 +1228,71 @@ int pjsip_acf_session_refresh_write(struct ast_channel *chan, const char *cmd, c
 
 	return ast_sip_push_task_wait_serializer(channel->session->serializer, refresh_write_cb, &rdata);
 }
+
+struct hangup_data {
+	struct ast_sip_session *session;
+	int response_code;
+};
+
+static int pjsip_hangup(void *obj)
+{
+	struct hangup_data *hdata = obj;
+	pjsip_tx_data *packet = NULL;
+
```

Comment:
The hangup cause on the channel should be a conversion from SIP response code to ISDN cause code for a bit of consistency

Reply:
It's set by chan_pjsip_session_end when the session ends.